src/tests/pools/D3M4626TypePool.t.sol

```diff
@@ -0,0 +1,168 @@
+
+        setPoolContract(pool = new D3M4626TypePool(ILK, address(hub), address(dai), address(vault)));
+
+        dai.approve(address(vault), type(uint256).max);
+    }
+
+    function invariant_dai_value() public {
+        assertEq(address(pool.dai()), address(dai));
+    }
+
+    function invariant_vault_value() public {
+        assertEq(address(pool.vault()), address(vault));
+    }
+
+    function invariant_vat_value() public {
+        assertEq(address(pool.vat()), address(vat));
+    }
+
+    function invariant_ilk_value() public {
+        assertEq(pool.ilk(), ILK);
+    }
+
+    function test_cannot_file_hub_no_auth() public {
+        pool.deny(address(this));
+        vm.expectRevert("D3M4626TypePool/not-authorized");
+        pool.file("hub", address(123));
+    }
+
+    function test_deposit_calls_vault_deposit() public {
+        deal(address(dai), address(pool), 1);
+        vm.prank(address(hub)); pool.deposit(1);
+        
+        assertEq(pool.assetBalance(), 1);
+        assertEq(dai.balanceOf(address(pool)), 0);
+    }
+
+    function test_withdraw_calls_vault_withdraw() public {
+        deal(address(dai), address(pool), 1);
+        vm.prank(address(hub)); pool.deposit(1);
+        
+        vm.prank(address(hub)); pool.withdraw(1);
+        
+        assertEq(pool.assetBalance(), 0);
+        assertEq(dai.balanceOf(address(hub)), 1);
+    }
+
+    function test_withdraw_calls_vault_withdraw_vat_caged() public {
+        deal(address(dai), address(pool), 1);
+        vm.prank(address(hub)); pool.deposit(1);
+        
+        vat.cage();
+        vm.prank(address(hub)); pool.withdraw(1);
+
+        assertEq(pool.assetBalance(), 0);
+        assertEq(dai.balanceOf(address(hub)), 1);
+    }
+
+    function test_redeemable_returns_vault() public {
+        assertEq(pool.redeemable(), address(vault));
+    }
+
+    function test_exit_vault_tokens() public {
+        deal(address(dai), address(this), 1e18);
+        vault.deposit(1e18, address(this));
+        uint256 tokens = vault.totalSupply();
+        vault.transfer(address(pool), tokens);
+        assertEq(vault.balanceOf(address(this)), 0);
+        assertEq(vault.balanceOf(address(pool)), tokens);
+
+        end.setArt(tokens);
+        vm.prank(address(hub)); pool.exit(address(this), tokens);
+
+        assertEq(vault.balanceOf(address(this)), tokens);
+        assertEq(vault.balanceOf(address(pool)), 0);
+    }
+
+    function test_quit_moves_balance() public {
+        deal(address(dai), address(this), 1e18);
+        vault.deposit(1e18, address(this));
+        vault.transfer(address(pool), vault.balanceOf(address(this)));
+        assertEq(vault.balanceOf(address(this)), 0);
+        assertEq(vault.balanceOf(address(pool)), 1e18);
+
+        pool.quit(address(this));
+
+        assertEq(vault.balanceOf(address(this)), 1e18);
+        assertEq(vault.balanceOf(address(pool)), 0);
+    }
+
+    function test_assetBalance_gets_vault_balanceOf_pool() public {
```

Comment:
These tests would be better if we could change the ratio of the total assets deposited again the vault supply.

Reply:
I might be getting confused though, but seeing that we use `convertToAssets` for getting `assetBalance` makes me almost sure I'm right.

Reply:
Re-opening so @sunbreak1211 can confirm this is complete.

Reply:
I think this is resolved. @sunbreak1211